clippy_lints/src/matches.rs

```diff
@@ -1118,7 +1118,7 @@ fn check_wild_enum_match(cx: &LateContext<'_>, ex: &Expr<'_>, arms: &[Arm<'_>])
 
     match missing_variants.as_slice() {
         [] => (),
-        [x] if !adt_def.is_variant_list_non_exhaustive() && !is_doc_hidden(cx, x) => span_lint_and_sugg(
+        [x] if !adt_def.is_variant_list_non_exhaustive() => span_lint_and_sugg(
```

Comment:
Your implementation effected the emission of the `match_wildcard_for_single_variants` lint. I'm not sure if the lint could now maybe be triggered by something like this:

```rs
enum AlphabetEnum {
    A,
    B,
    C,
    #[doc(hidden)]
    __Private,
}
match Enum::A {
    AlphabetEnum::A => (),
    AlphabetEnum::B => (),
    _ => (),
}
```

Could you add a test for that case :)

Reply:
Great catch!